### DISCUSSION TITLE
Blazor wasm?

### DISCUSSION THREAD

**Discussion by jonmiller1** [dlemstra/magick-wasm#35] [General]
I’ve used the Magick.net library and it is great. This new wasm version is very interesting. I’d like to find a way to use the wasm from blazor. It would be even better if it matched the magick.net api. Is this possible?

**Comment by dlemstra** [comment]
I try to match the API where possible but this library does not contain all the methods of Magick.NET yet. This is a typescript/javascript library that you can use on the client side or server side. Not sure what your use case is though.

**Reply by jonmiller1** [reply]
I'm hosting an app on a raspberry pi and I'd like to offload some of the work in the browser.  Specifically, I want to convert a 256 color gray image to a binary using the OTSU binarization.  Can I do that with javascript and the magik-wasm?

**Comment by dlemstra** [comment]
It seems that you will need to use the `AutoThreshold` method for that and this is not yet support in the wasm library but I could add this. And then you will be able to use that method. Is that all that you are using of Magick.NET?

**Reply by jonmiller1** [reply]
For now it would be.  It would be nice if I do things like rotate and scale images as well.  However, just converting a gray to binary would be huge for me.

**Comment by dlemstra** [comment]
I just published a new release that includes the `autoThreshold` method for you.

**Reply by jonmiller1** [reply]
Thank you so much! I'm going to try it out.
